failed to parse application from Clusterfile

### What happen?

sealer apply -f xxx failed

### Relevant log output?

```Shell
2023-04-04 17:35:28 [ERROR] [root.go:75] sealer-v0.9.3: failed to parse application from Clusterfile:app name `a` is not found in [b c d]
```


### What you expected to happen?

new cluster should be created successfully

### How to reproduce it (as minimally and precisely as possible)?

```YAML
1. a, b, c, d are 4 apps defined in kubefile, and b c d are launched by default
2. define a clusterfile with appNames: [a, b, c, d]
3. run sealer apply -f xxx
```


### Anything else we need to know?

_No response_

### What is the version of Sealer you using?

0.9.3

### What is your OS environment?

_No response_

### What is the Kernel version?

_No response_

### Other environment you want to tell us?

- Cloud provider or hardware configuration:
- Install tools:
- Others:

--- a/pkg/application/v2app.go
+++ b/pkg/application/v2app.go
@@ -165,10 +165,13 @@ func NewV2Application(app *v2.Application, extension imagev1.ImageExtension) (In
 		app:                 app,
 		extension:           extension,
 		globalCmds:          extension.Launch.Cmds,
-		launchApps:          extension.Launch.AppNames,
 		appLaunchCmdsMap:    map[string][]string{},
 		appRootMap:          map[string]string{},
 		appFileProcessorMap: map[string][]FileProcessor{},
+	}
+
+	for _, registered := range extension.Applications {
+		v2App.launchApps = append(v2App.launchApps, registered.Name())
 	}
 
 	// initialize globalCmds, overwrite default cmds from image extension.

